pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -153,6 +153,13 @@ public synchronized void addConsumer(Consumer consumer) throws BrokerServiceExce
             throw new ConsumerBusyException("Subscription reached max consumers limit");
         }
 
+        if (consumerList.contains(consumer)) {
```

Comment:
Can you provide the case of duplicated consumer? 
I wonder if we should replace the old consumer with the new one. The older consumer maybe somewhat expired.

Reply:
Yes so I think we should write a method like org.apache.pulsar.broker.service.Producer#isSuccessorTo to judge whether the previous consumer object should be overwritten by consumerEpoch，like this:

public boolean isSuccessorTo(Consumer other){
    return Objects.equals(cnx.clientAddress(), other.cnx.clientAddress()) && consumerId == other.consumerId
            &&  other.consumerEpoch < consumerEpoch;
}

Reply:
> Can you provide the case of duplicated consumer?

OK, I will try to find duplicate cases.